test/dlc_tests/conftest.py

```diff
@@ -361,14 +377,22 @@ def pytest_generate_tests(metafunc):
                 if lookup in image:
                     is_example_lookup = "example_only" in metafunc.fixturenames and "example" in image
                     is_standard_lookup = "example_only" not in metafunc.fixturenames and "example" not in image
+                    tf2_requirement_failed = "tf2_only" in metafunc.fixturenames and is_tf1(image)
```

Comment:
if we need to add more tf specific conditions like this, will we have to bloat this function even further? I suggest having a separate function to handle any tf2 specific requirements.

Additionally, why do we have functions like is_tf20 instead of having functions that are is_tf_version(image, tf_version) - in terms of scaling for future versions it becomes cumbersome to create a new function every time

Reply:
I actually don't know why those were added. I can try to clean this whole set of fixtures up, and make the TF 2 fixtures generic.

Reply:
please focus on just cleaning up the tf logic that was added so as not to outscope this PR